contracts/bonding/BondingManager.sol

```diff
@@ -224,32 +219,38 @@ contract BondingManager is IBondingManager, Controllable {
             transcoderPools.addTranscoder(_to, stakeForTranscoder);
         }
 
+        if (_amount > 0) {
+            // Only transfer tokens if _amount is greater than 0
+            // Transfer the token. This call throws if it fails.
+            token.transferFrom(msg.sender, this, _amount);
+        }
+
         return true;
     }
 
-    /**
-     * @dev Unbond your current stake. You will enter the unbonding phase for
-     * the unbondingPeriod.
+    /*
+     * @dev Unbond delegator's current stake. Delegator enters unbonding state
      */
     function unbond() external returns (bool) {
-        // Check if current round is initialized
-        if (!roundsManager().currentRoundInitialized()) throw;
-        // Check if this is an initialized delegator
-        if (delegators[msg.sender].initialized == false) throw;
```

Comment:
If the sender passes this check `require(delegatorStatus(msg.sender) == DelegatorStatus.Bonded);` `delegator[msg.sender].initialized` cannot be `false` because the only times it is `false` is if the delegator is not registered (i.e. never bonded to a transcoder, or has unbonded and withdrawn) and if that is the case, `delegator[msg.sender].startRound` would be 0 so the sender wouldn't pass that first check